extension/sql/timescale-prometheus.sql

```diff
@@ -0,0 +1,85 @@
+
+CREATE OR REPLACE FUNCTION const_support(internal) RETURNS INTERNAL
+AS '$libdir/timescale_prometheus_extra', 'const_support'
+LANGUAGE C IMMUTABLE STRICT;
+
+--wrapper around jsonb_each_text to give a better row_estimate
+--for labels (10 not 100)
+CREATE OR REPLACE FUNCTION label_jsonb_each_text(js jsonb,  OUT key text, OUT value text)
+ RETURNS SETOF record
+ LANGUAGE internal
+ IMMUTABLE PARALLEL SAFE STRICT ROWS 10
+AS $function$jsonb_each_text$function$;
+
+--wrapper around unnest to give better row estimate (10 not 100)
+CREATE OR REPLACE FUNCTION label_unnest(label_array anyarray)
+ RETURNS SETOF anyelement
+ LANGUAGE internal
+ IMMUTABLE PARALLEL SAFE STRICT ROWS 10
+AS $function$array_unnest$function$;
+
+---------------------  comparison functions ---------------------
+
+CREATE OR REPLACE FUNCTION label_find_key_equal(label_key prom.label_key, pattern prom.pattern)
+RETURNS prom.matcher_positive
```

Comment:
I find it confusing where we use the templating thing with the different schemas and where we don't. Is it that we can't run the matching thingy over these files? Or is it something else, it just strikes me that the mixture of both strategies is odd and could easily lead to errors in the future.

Reply:
This is a good point. this shouldn't be schema qualified.

Reply:
ok, should work now that they're in the path

Reply:
Nope, doesn't work. I think the way that the search path is set on `CREATE EXTENSION` prevents it.

Reply:
Ok, I got it to work, but the extension will only be relocatable at install